src/screens/TopicPageScreen/MemoizedListComponent.js

```diff
@@ -154,6 +158,9 @@ const RenderListFeed = (props) => {
           hideThreeDot={true}
           props={item}
           height={getHeightHeader()}
+          isSelf={item?.is_self}
+          isFollow={item?.is_following_target}
+          onPressFollUnFoll={() => followUnfollowTopic(item, route?.params?.id, offset)}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `isSelf` and `isFollow` props are added to the Header component, and the `onPressFollUnFoll` prop now calls the `followUnfollowTopic` function. Ensure these changes are reflected in the PropTypes of the Header component and that they don't break existing functionality.

223:
The `isShowDM` prop is added to the Footer component. Make sure this change is reflected in the PropTypes of the Footer component and doesn't break existing functionality.


<!-- This is an auto-generated comment by OSS CodeRabbit -->